lang/syn/src/parser/accounts/constraints.rs

```diff
@@ -512,16 +530,10 @@ impl<'ty> ConstraintGroupBuilder<'ty> {
                             )),
                         },
                     }
-                } else if let Some(tm) = &associated_token_mint {
+                } else if let Some(at) = &associated_token {
                     InitKind::AssociatedToken {
-                        mint: tm.clone().into_inner().mint,
-                        owner: match &associated_token_authority {
-                            Some(a) => a.clone().into_inner().auth,
-                            None => return Err(ParseError::new(
-                                tm.span(),
-                                "authority must be provided to initialize a token program derived address"
```

Comment:
One thing we changed was that this code only prints an error if the authority is missing, but the symmetric situation of having an authority but missing a mint doesn't print symmetrically—you just skip the entire `if let Some`.